### ISSUE TITLE
Unexpected token: self

### ISSUE THREAD

**Issue by egircys** [pdepend/pdepend#192]
- PDepend 2.0.6
- PHP 5.6.6-1~dotdeb.1

``` php
<?php

class Book
{
    const A = 0;
    const B = 1;

    const C = [self::A, self::B];
}
```

```
bin/pdepend --summary-xml=pdepend.xml Book.php
```

```
PDepend 2.0.6

Parsing source files:
.                                                                1

Calculating Cyclomatic Complexity metrics:
.     0

Calculating Node Loc metrics:
.     0

Calculating NPath Complexity metrics:
.     0

Calculating Inheritance metrics:
.     0

Calculating Node Count metrics:
.     0

Calculating Hierarchy metrics:
.     0

Calculating Code Rank metrics:
.     0

Calculating Coupling metrics:
.     0

Calculating Class Level metrics:
.     0

Calculating Cohesion metrics:
.     0

Generating pdepend log files, this may take a moment.

The following error occured:
Unexpected token: self, line: 8, col: 25, file: /var/www/symfony/Book.php.


Time: 0:00:00; Memory: 5.50Mb

```

**Comment by hackel** [comment]
I'm having the same isse, but in my case, I have:
```php
	const A = [
		'B',
		'C',
	];
```
I get `Unexpected token: 'C'`.  I believe this must be an incompatibilty with arrays defined as aconstants in general, not specifically related to using `self::`.

**Comment by internalsystemerror** [comment]
For reference, or for someone as a starting point, the fix looks like it may need to be applied somewhere in [src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php] (https://github.com/pdepend/pdepend/blob/master/src/main/php/PDepend/Source/Language/PHP/AbstractPHPParser.php#L5985) around line 6000 (ish). There are two methods around there which parse constants.

**Comment by sustmi** [comment]
I tried to fix the issue: https://github.com/pdepend/pdepend/compare/master...sustmi:issue-192-array-in-class-constant
but I have a problem with existing test that is now failing: https://github.com/pdepend/pdepend/blob/master/src/test/resources/files/issues/079/testParserThrowsExpectedExceptionForArrayInConstantDeclaration.php .

How does PDepend handle different PHP versions? What about features such this that were not possible in previous PHP versions (pre 5.6) and are possible now?

Comment of ```PHPParserGeneric``` class says:
> Concrete parser implementation that is very tolerant and accepts language constructs and keywords that are reserved in newer php versions, but not in older versions.

Does that mean that PDepend should always supports all the features of the latest PHP version?
If it is so then I guess that the test should be deleted.

**Comment by manuelpichler** [comment]
Fixed in 1209b0e

**Comment by sustmi** [comment]
Thank you, @manuelpichler .
